src/event_loop.rs

```diff
@@ -88,7 +88,10 @@ pub enum ControlFlow {
     /// Send a `LoopDestroyed` event and stop the event loop. This variant is *sticky* - once set,
     /// `control_flow` cannot be changed from `Exit`, and any future attempts to do so will result
     /// in the `control_flow` parameter being reset to `Exit`.
-    Exit,
+    /// The contained number will be used as exit code, if the platform supports that (this
+    /// _excludes_ ios, android and wasm32), it's an [`u8`] to prevent some weird surprises as
+    /// described in [`std::process::exit`].
+    Exit(u8),
```

Comment:
Variant 2 looks better to me, since variant 1 lets in my eyes

1. fall `From` implementations out-of-date (if a dependency upgrades winit with this, sees `#[non_exhaustive]`, adds a `_ => Self::Unknown` arm, and upgrades winit later on with another new variant in there, then this could be easily forgotten and whoops, suddently we have something that impls `From<ControlFlow>` but doesn't map the new variant correctly).
2. seem `Exit` like it wouldn't exit without any code at all, which isn't true.

Reply:
I prefer the first alternative, as I don't think anyone's really `match`-ing on `ControlFlow`.

>2. seem `Exit` like it would exit without any code at all, which isn't true.

I don't think this is an issue. If someone cares about the exit code, then I think we can trust them to assume that `0` will be used in the absence of any errors (like panics). In any case, we'd want to document exactly what `ControlFlow::Exit` does.

Reply:
Then I'm okay with the `ExitWithCode` variant co-existing with the `Exit` variant, even though I'm not too sure about `#[non_exhaustive]`. What exact advantage does that add if no one seriously matches `ControlFlow`?

Reply:
It's mostly to make (potential) future additions be a non-breaking change.

Reply:
> I prefer the first alternative, as I don't think anyone's really `match`-ing on `ControlFlow`.

Then the associated constant would work just as well, in terms of breakage, right?

I think my preference would be the associated constant, since our code won't have to handle two enum variants that does effectively the same thing.